addons/stock_negative/stock_negative.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, api, _
+from openerp.exceptions import except_orm
+
+
+class stock_quant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def _quant_create(self, qty, move, lot_id=False, owner_id=False, src_package_id=False, dest_package_id=False, force_location_from=False, force_location_to=False):
+        quant = {}
+        if move.location_id.usage == 'internal':
```

Comment:
No way, I clearly stated in my previous comments that qty_available should
not be checked here.  It will create quants if can not find any to move, so
it will create negative => raise.

On Thu, Jan 8, 2015 at 5:05 PM, Richard Mathot <notifications@github.com>
wrote:

> In addons/stock_negative/stock_negative.py
> <https://github.com/odoo/odoo/pull/4318#discussion-diff-22660230>:
>
> > @@ -0,0 +1,17 @@
> > +# -*- coding: utf-8 -*-
> > +
> > +from openerp import models, api, _
> > +from openerp.exceptions import except_orm
> > +
> > +
> > +class stock_quant(models.Model):
> > +    _inherit = "stock.quant"
> > +
> > +    @api.model
> > +    def _quant_create(self, qty, move, lot_id=False, owner_id=False, src_package_id=False, dest_package_id=False, force_location_from=False, force_location_to=False):
> > +        quant = {}
> > +        if move.location_id.usage == 'internal':
>
> don't you also need to test that product_qty_available - qty < 0 ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/odoo/odoo/pull/4318/files#r22660230>.
>